spec/models/context_spec.rb

```diff
@@ -0,0 +1,38 @@
+require 'spec_helper'
+
+RSpec.describe Context do
+  describe 'initialization' do
+    it 'allows keyword init' do
+      context = Context.new user: double(:user)
+      expect(context.user).to be_present
+    end
+
+    it 'allows args to be omitted' do
+      context = Context.new
+      expect(context.user).to be_nil
+    end
+  end
+
+  describe 'lazy, memoized, overridable readers' do
+    let(:current_settings) { double :current_settings }
+    let(:system_setting) { class_double('SystemSetting').as_stubbed_const }
+    let(:context) { Context.new }
+
+    it 'is lazy' do
+      expect(system_setting).to receive(:current_settings) { current_settings }
+      expect(context.system_settings).to be current_settings
```

Comment:
Maybe because I'm a little rusty with the RSpec matchers here, I'm not sure how this definitively tests the laziness. A non-lazy implementation wouldn't pass this test?

Reply:
Good point! I think i was making the mental leap that since we didn't pass a `SystemSetting` into the `Context` initializer, then verifying the getter returns the current one _infers_ a lazy lookup. But that's all just in my head :)

I'll change this to make the expectation and method of verification clearer.

Reply:
Attempted to clarify in 1bac57d.